custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -327,6 +338,93 @@ def handle_exception(self, exception, repeat_record):
                                 "followup_nikshay_registered", "followup_nikshay_error")
 
 
+class NikshayRegisterPrivatePatientPayloadGenerator(BaseNikshayPayloadGenerator):
+    format_name = 'case_xml'
+    format_label = 'XML'
+
+    @memoized
```

Comment:
what if you just added:
```py
@memoized
def _get_person_locations(self, person_case):
    return person_locations(person_case)
```
to this payload generator? 
Then in your `handle_success` method you'd do:
```py
health_facility_id = self._get_person_locations(person_case).pcp
```
That would serve the same purpose and be a bit less confusing. 

If you go that route, you could also cache the `get_person_from_episode` call in the same way.